test/fn-api-tests/utils.go

```diff
@@ -71,31 +97,43 @@ func getServerWithCancel() (*server.Server, context.CancelFunc) {
 
 		s = server.New(ctx, server.WithDBURL(dbURL), server.WithMQURL(mqURL), server.WithFullAgent())
 
-		go s.Start(ctx)
-		started := false
-		time.AfterFunc(time.Second*10, func() {
-			if !started {
-				panic("Failed to start server.")
-			}
-		})
-		_, err := http.Get(apiURL + "/version")
-		for err != nil {
-			_, err = http.Get(apiURL + "/version")
-		}
-		started = true
-		cancel2 = context.CancelFunc(func() {
-			cancel()
+		go func() {
+			s.Start(ctx)
 			os.Remove(tmpMq)
 			os.Remove(tmpDb)
-		})
+		}()
+
+		startCtx, startCancel := context.WithDeadline(ctx, time.Now().Add(time.Duration(10)*time.Second))
```

Comment:
i think we want this at the head of `startServer` and to be `ctx, cancel := context.WithDeadline`, not sure we need both contexts in a separate (pardon me) context since checking the server is directly related to starting the server.

Reply:
i see, we want to leave `s.Start` up but only timeout the check. ignore me.